From 2edab935ca49e6e628143975a6654dbd3d78f51c Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Tue, 23 Feb 2021 11:29:55 -0700
Subject: [PATCH 1743/1851] mtd: spi-nor: Fix call trace when qspi driver built
 as module

commit 700725af75ecddb674ccadf3f6d06b9bd5df79a5 from
https://github.com/Xilinx/linux-xlnx.git

When spi-nor and zynqmp qspi driver are built as modules, call trace was
observed while loading the modules. With this fix call trace issue is resolved.

Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Acked-by: Tony McDowell <tmcdowe@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index 12d1b61350aa..678eaae011d0 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -2484,7 +2484,7 @@ static struct spi_nor_fixups gd25q256_fixups = {
  * For historical (and compatibility) reasons (before we got above config) some
  * old entries may be missing 4K flag.
  */
-static const struct flash_info spi_nor_ids[] = {
+static struct flash_info spi_nor_ids[] = {
 	/* Atmel -- some are (confusingly) marketed as "DataFlash" */
 	{ "at25fs010",  INFO(0x1f6601, 0, 32 * 1024,   4, SECT_4K) },
 	{ "at25fs040",  INFO(0x1f6604, 0, 64 * 1024,   8, SECT_4K) },
-- 
2.31.1

